### PR TITLE
PXP-6895 NIAID CT bundle

### PR DESCRIPTION
**Pull request by mfshao**
Jira Ticket: [PXP-6895](https://ctds-planx.atlassian.net/browse/PXP-6895)

Separate NIAID Clinical Trials related code into a dedicated bundle

Deployed in https://mingfei.planx-pla.net/ with `GEN3_BUNDLE=nct`

### New Features
- Separate NIAID Clinical Trials related code into a dedicated bundle

### Bug Fixes
- Various warning and crash fixes

### Improvements
- Use antd `Carousel` component instead of `react-slick`
- Index page chart carousel now support adding static images
- Index page introduction text now supports `multiLineTexts` and embedded HTML tags

### Deployment changes
- New value for `GEN3_BUNDLE` env var `GEN3_BUNDLE=nct`
- New optional portal config variable `injectDAPTag` (default value is `false`). If set to `true`, will inject DAP tag script into page header (this is required for NCT PROD env only)

### CODE REVIEW

File: src/index.ejs [hunk 1]
```diff
@@ -11,11 +11,15 @@
...
     <link href="https://fonts.googleapis.com/icon?family=Source+Sans+Pro" rel="stylesheet">
+    <link href="https://fonts.googleapis.com/css2?family=Roboto&display=swap" rel="stylesheet">
+    <link href="https://fonts.googleapis.com/css2?family=Martel&display=swap" rel="stylesheet">
     <link rel="stylesheet/less" type="text/css" media="all" href="<%= htmlWebpackPlugin.options.basename %>/node_modules/@gen3/ui-component/dist/css/base.less">
     <link rel="stylesheet" type="text/css" media="all" href="<%= htmlWebpackPlugin.options.basename %>/src/css/graphiql.css" />
     <link rel="icon"
     type="image/png" href="<%= htmlWebpackPlugin.options.basename %>/src/img/favicon.ico">
     <title><%= htmlWebpackPlugin.options.title %></title>
+    <!-- We participate in the US government's analytics program. See the data at analytics.usa.gov. -->
+    <script async type="text/javascript" src="https://dap.digitalgov.gov/Universal-Federated-Analytics-Min.js?agency=NIH&subagency=NIAID" id="_fed_an_ua_tag"></script>
```
Comment (mfshao) [inline src/index.ejs:22]: this is temporary

i will use `htmlWebpackPlugin` to inject this only for NIAID once i separate this NIAID thing into a dedicated GEN3_BUNDLE

File: src/Covid19Dashboard/ChartCarousel/ChartCarousel.less [hunk 2]
```diff
@@ -42,3 +42,44 @@
   white-space: pre-wrap;
   z-index: 10; // place it over the "mapbox" link...
 }
+
+/** override some style for carousel's arrow and dots **/
```
Comment (mfshao) [inline src/Covid19Dashboard/ChartCarousel/ChartCarousel.less:46]: i'm thinking maybe later we can refactor the carousel component in covid19page into using `antd carousel` as wel

File: src/Covid19Dashboard/PlotChart/index.jsx [hunk 3]
```diff
@@ -47,13 +50,19 @@ class PlotChartAxisTick extends React.Component {
 }
 
 PlotChartAxisTick.propTypes = {
-  x: PropTypes.number.isRequired,
-  y: PropTypes.number.isRequired,
-  payload: PropTypes.object.isRequired,
+  x: PropTypes.number,
+  y: PropTypes.number,
+  payload: PropTypes.object,
```
Comment (mfshao) [inline src/Covid19Dashboard/PlotChart/index.jsx:55]: to fix some react warnings

File: src/index.ejs [hunk 4]
```diff
@@ -11,11 +11,17 @@
...
     <meta name="viewport" content="width=device-width" />
     <link href="https://fonts.googleapis.com/icon?family=Source+Sans+Pro" rel="stylesheet">
+    <link href="https://fonts.googleapis.com/css2?family=Roboto&display=swap" rel="stylesheet">
+    <link href="https://fonts.googleapis.com/css2?family=Martel&display=swap" rel="stylesheet">
     <link rel="stylesheet/less" type="text/css" media="all" href="<%= htmlWebpackPlugin.options.basename %>/node_modules/@gen3/ui-component/dist/css/base.less">
     <link rel="stylesheet" type="text/css" media="all" href="<%= htmlWebpackPlugin.options.basename %>/src/css/graphiql.css" />
     <link rel="icon"
     type="image/png" href="<%= htmlWebpackPlugin.options.basename %>/src/img/favicon.ico">
     <title><%= htmlWebpackPlugin.options.title %></title>
+    <% if(htmlWebpackPlugin.options.dap_tag) { %>
```
Comment (mfshao) [inline src/index.ejs:21]: for injecting NIAID DAP tag script, see `webpack.config.js`

**Review by mfshao — COMMENTED** [review]
I took out some CSS changes that are only intended for NCT envs. Will add them back via `gitops.css` for NCT QA and PROD envs

**Comment by mfshao** [comment]
example of required deployement changes https://github.com/uc-cdis/gitops-qa/pull/970

**Review by ZakirG — APPROVED** [review]

File: src/Index/utils.js [hunk 5]
```diff
@@ -85,6 +85,9 @@ export const mergeChunkedChartData = (chartDataArray) => {
 // request will fail for logged-out users. If the request fails because the user
 // is logged out, this function will return {needsLogin: true}.
 export const loadHomepageChartDataFromDatasets = async (callback) => {
+  if (homepageChartNodes.length === 0) {
```
Comment (paulineribeyre) [inline src/Index/utils.js:88]: 🙌

File: src/Covid19Dashboard/ChartCarousel/ChartCarousel.less (hunk 2 above)
Comment (paulineribeyre) [inline src/Covid19Dashboard/ChartCarousel/ChartCarousel.less:46]: should we deploy this portal branch to qa-covid19 to test that the carousels are fine?

Comment (mfshao) [reply src/Covid19Dashboard/ChartCarousel/ChartCarousel.less:46]: Good idea. I have tested locally against PRC and i think the carousels looks fine to me.

But I agree to deploy this to qa-covid19 so we can have more eyes on this to make sure I didn't miss anything

Comment (mfshao) [reply src/Covid19Dashboard/ChartCarousel/ChartCarousel.less:46]: https://github.com/uc-cdis/gitops-qa/pull/986 @paulineribeyre

**Pull request merged**